### PR TITLE
[native_toolchain_c] Use 16 kb pages for Android by default

### PR DESCRIPTION
**Pull request by dcharkes**
Closes: https://github.com/dart-lang/native/issues/1611

### CODE REVIEW

File: pkgs/native_toolchain_c/test/cbuilder/cbuilder_cross_android_test.dart [hunk 1]
```diff
@@ -67,6 +67,9 @@ void main() {
                 .firstWhere((e) => e.contains('file format'));
             expect(machine, contains(objdumpFileFormat[target]));
           }
+          if (linkMode == DynamicLoadingBundled()) {
+            await expectPageSize(libUri, 16 * 1024);
```
Comment (mkustermann) [inline pkgs/native_toolchain_c/test/cbuilder/cbuilder_cross_android_test.dart:71]: Do we have evidence of users running into this issue?

Not all users will use `package:native_toolchain_c`, some may ship binaries, etc. So we could add build output validation for this that will apply to all code assets.

```
if (targetOS == OS.android && androidVersion >= ...) {
  ensurePageSizeIs16Kb(...);
}
```

somewhere in https://github.com/dart-lang/native/blob/ee2215e57292c90de302a56df7c8e42d09a47509/pkgs/native_assets_cli/lib/src/code_assets/validation.dart#L102

**Review by mkustermann — APPROVED** [review]

File: pkgs/native_toolchain_c/test/cbuilder/cbuilder_cross_android_test.dart (hunk 1 above)
Comment (dcharkes) [reply pkgs/native_toolchain_c/test/cbuilder/cbuilder_cross_android_test.dart:71]: > Do we have evidence of users running into this issue?

* `plugin_ffi` users were running into this: https://github.com/flutter/flutter/pull/155508
* I'm running into this locally with the latest version of the emulator on Linux and MacOS.

> So we could add build output validation for this that will apply to all code assets.

We're not mandating `readelf` or `objdump` currently. Flutter provides access to a compiler, linker, and archiver via `cCompiler`. But we don't have something similar for commandline utilities at the moment.

Comment (dcharkes) [reply pkgs/native_toolchain_c/test/cbuilder/cbuilder_cross_android_test.dart:71]: > `androidVersion`

We don't have an android version. Only a minAndroidSdkVersion. But the minAndroidSdk doesn't matter, something compiled with a lower min SDK trying to run on Android 15 will still fail to load the dylib if the page-size is smaller than 16kb.

**Comment by dcharkes** [comment]
Thanks @mkustermann!

I've filed https://github.com/dart-lang/native/issues/1741 to add the command-line utilities somewhere so that we can rely on them and report users via a doctor command if the utilities are not there.

**Pull request merged**